src/build/__init__.py

```diff
@@ -142,6 +142,7 @@ def check_dependencies(self, distribution):  # type: (str) -> Set[str]
         :param distribution: Distribution to build (sdist or wheel)
         '''
         dependencies = self.get_dependencies(distribution)
+        dependencies.update(self.build_dependencies)
```

Comment:
This will fix issue #64, but it occurs to me that `self.get_dependencies(distribution)` will actually *fail* in the case where a dependency required to execute the PEP 517 hooks is missing.

I think this may be a fundamental problem with the `check_dependencies` API, though. You can't actually get the full set of missing dependencies if a subset of those dependencies are missing. It might be worth splitting this into two functions like `check_build_system_requires` and `check_build_artifact_dependencies`.

Another option would be to document that this doesn't return *all* the missing dependencies if the build dependencies aren't satisfied, but the problem there is that the caller has no way of knowing where it failed.

Reply:
I'll note that this is a downside of the mocking approach you've taken here. We would have caught this issue if you were running a real build, because the mock is returning an empty set even when in reality it would raise a missing backend error.